### PR TITLE
feat: Use publishBundle to deploy contract to chain

### PR DESCRIPTION
**Pull request by kriskowal**
Take advantage of publishBundle power to send the contract more directly to the connected chain, instead of posting a large message.

Using the new feature https://github.com/Agoric/agoric-sdk/pull/4975

In partial fulfillment of https://github.com/Agoric/agoric-sdk/issues/4564

### CODE REVIEW

**Review by samsiegart — APPROVED** [review]
Neat!

File: contract/deploy.js [hunk 1]
```diff
@@ -17,27 +17,31 @@ import bundleSource from '@endo/bundle-source';
...
+const installBundle = async (powers, zoe, board) => {
+  const { pathResolve, publishBundle } = powers;
   // We must bundle up our contract code (./src/contract.js)
   // and install it on Zoe. This returns an installationHandle, an
   // opaque, unforgeable identifier for our contract code that we can
   // reuse again and again to create new, live contract instances.
   const bundle = await bundleSource(pathResolve(`./src/contract.js`));
-  const installation = await E(zoe).install(bundle);
+
+  const hashBundle = await publishBundle(bundle);
```
Comment (mhofman) [inline contract/deploy.js]: Nit / bikeshed: `hashBundle` sound like a function. Maybe `hashedBundle` or `bundleHash` ?

File: contract/deploy.js [hunk 2]
```diff
@@ -17,27 +17,31 @@ import bundleSource from '@endo/bundle-source';
 /**
  * @typedef {Object} DeployPowers The special powers that agoric deploy gives us
```
Comment (mhofman) [inline contract/deploy.js:18]: I wish these were imported instead, but we have a bit of work on typing to do before we get there.

File: contract/deploy.js [hunk 3]
```diff
@@ -60,7 +64,7 @@ const installBundle = async (pathResolve, zoe, board) => {
  * Object, wallet: ERef<Object>, faucet: ERef<Object>}>} homePromise
  * @param {DeployPowers} powers
  */
-const deployContract = async (homePromise, { pathResolve }) => {
+const deployContract = async (homePromise, { pathResolve, publishBundle }) => {
```
Comment (mhofman) [inline contract/deploy.js:67]: Quick question, will older SDKs simply not provide `publishBundle` in the `deployPowers`? In which case the loadgen could feature test for presence and fallback to installing the bundle directly if it's missing?

Comment (kriskowal) [reply contract/deploy.js:67]: Yes, `publishBundle` is absent in prior versions of `agoric-cli`.

**Pull request merged**